### PR TITLE
Implement /get_prefs and /save_prefs API 

### PR DESCRIPTION
**Pull request by cindyli**
Resolves https://github.com/fluid-project/personal-data-server/issues/5

This pull request implements /get_prefs and /save_prefs API. Other endpoints mentioned in https://github.com/fluid-project/personal-data-server/issues/5 have already implemented and merged.

### CODE REVIEW

File: src/server/routes/prefs.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+ */
+router.post("/save_prefs", async function (req, res) {
+    const loginToken = getLoginToken(req.headers.authorization);
+
+    if (!loginToken) {
+        res.status(403).json({"isError": true, "message": "Please login first."});
+        return;
+    }
+
+    if (!req.body) {
```
Comment (amb26) [inline src/server/routes/prefs.js:64]: This could be achieved by setting the "strict" option of the body parser

File: src/server/routes/prefs.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    if (isSuccess) {
+        res.send("Saved successfully.");
+    } else {
+        res.status(403).json({"isError": true, "message": "Cannot save the incoming preferences."});
+        return;
+    }
+});
+
+/**
+ * Parse the login token.
```
Comment (amb26) [inline src/server/routes/prefs.js:96]: Give more details about what values for the header are considered valid

File: tests/prefsApiTests.js [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+        jqUnit.assertTrue("The database docker container has been stopped", response.dbStopped);
+    }
+
+    // Stop the server
+    await server.stopServer(serverInstance.server);
+    serverStatus = await fluid.personalData.getServerStatus(config.server.port);
+    jqUnit.assertFalse("The server has been stopped", serverStatus);
+});
+
+// create a large object: https://gist.github.com/artifi/2047874
```
Comment (amb26) [inline tests/prefsApiTests.js:208]: I think this strategy would be more direct: https://stackoverflow.com/a/34811384 e.g.
{
     key: "#".repeat(10240);
}

**Comment by cindyli** [comment]
@amb26, all review comments have been addressed. Ready for another review. Thanks.

File: src/server/routes/prefs.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+    // Verify if the login token is valid
+    const prevPrefs = await dbOps.getPreferences(loginToken);
+
+    if (!prevPrefs) {
+        res.status(403).json({"isError": true, "message": "Invalid login token. Please login."});
+        return;
+    }
+
+    const isSuccess = await dbOps.savePrefsByLoginToken(loginToken, req.body);
```
Comment (amb26) [inline src/server/routes/prefs.js:77]: Please add a comment at this line recording the discussion at the tech meeting - that is, that in the case that the client supplies an empty body, the JSON middleware will convert this into an empty object which will be saved, and link to the dodgy line https://github.com/expressjs/body-parser/blob/master/lib/types/json.js#L74 where this possibly surprising behaviour is implemented

**Pull request merged**